### ISSUE TITLE
Undo/Redo Keymap changes not working in script editor

### ISSUE THREAD

**Issue by Thane5** [godotengine/godot#50972]
### Godot version

3.3.2.stable

### System information

Windows 10, GLES3

### Issue description

As i use a QWERTZ keyboard, i wanted to remap undo/redo to ctrl+y and ctrl+x. This worked fine for the scene editor, but does not work in the script editor. The editor settings save my keymap, but it gets simply ignored in the script editor, and still uses ctrl+Z

### Steps to reproduce

1. remap "undo" to Ctrl+X
2. Try to undo anything in the script editor

**Comment by Calinou** [comment]
I think this is because the shortcuts are hardcoded in the TextEdit node, and I'm not sure if this can be fixed due to https://github.com/godotengine/godot/issues/29730. (TextEdit is also available for use in projects.)

**Comment by EricEzaM** [comment]
Yep, fixed by #43663 in master, but won't be backported to 3.x (at least not by me) due to the large set of changes and differences between 3.x and 4.x

**Comment by Thane5** [comment]
Interestingly i had a similar issue in Godot 4, where ctrl+x would always be mapped to "Cut", even though i set it to redo. But i guess i will create a new error report for that.
